### ISSUE TITLE
Pruning doesn't support 'Prelu'?

### ISSUE THREAD

**Issue by BlossomingL** [NervanaSystems/distiller#409]
Hi~ I pruned a model containing 'Prelu(64)', but it failed, would you please give me some suggestion?

**Comment by nzmora** [comment]
Hi @LinXXXX ,

Please provide more information - it's not clear what failed and how.  Logs would be very helpful.

Cheers,
Neta

**Comment by LinXXXX** [comment]
Thanks for your qiuck reply.
For example, the code:
`self.conv1 = nn.Conv2d(3, keep[0], kernel_size=3, stride=1, padding=1,
                               bias=False)
        self.bn2 = nn.BatchNorm2d(64, eps=2e-5, momentum=0.9)
        self.relu = nn.PReLU(64)`
In this code, there is a PReLU with a parameter 64 which correlates with prior out channel(64), when I prune prior conv1, PReLU should also be pruned, but I didn't find code to deal with it.

**Comment by nzmora** [comment]
Thanks @LinXXXX 

Indeed there is a dependency between the number of `self.conv1`'s output channels and the learned parameter of `self.relu` which we currently do not address.  The dependency is similar to the one between `self.conv1` and 'self.bn2'.
Are you only pruning or also performing thinning?  If you just prune w/o physically removing channels, you should not see a problem, but that might not be what you want.

Cheers 
Neta

**Comment by nzmora** [comment]
Excellent @LinXXXX ,

You can issue a PR with the fixed `thinning.py` code if you want us to integrate it to 'master'.  Please also add a test (e.g. show that [this](https://github.com/NervanaSystems/distiller/issues/409#issuecomment-546196398) network doesn't fail thinning).  

Thanks!
Neta
